src/azul/queues.py

```diff
@@ -108,6 +100,27 @@ def _dump_messages(self, messages, queue_url, path):
             json.dump(content, file, indent=4)
         logger.info('Wrote %i messages', len(messages))
 
+    def _get_messages(self, queue):
+        messages = []
+        while True:
+            message_batch = queue.receive_messages(AttributeNames=['All'],
+                                                   MaxNumberOfMessages=10,
+                                                   VisibilityTimeout=300)
+            if not message_batch:  # Nothing left in queue
+                break
+            else:
+                messages.extend(message_batch)
+        return messages
+
+    def read_messages(self, queue):
+        messages = self._get_messages(queue)
+        message_batches = list(more_itertools.chunked(messages, 10))
```

Comment:
This duplicates code in _dump()

Reply:
I added the changes to this file as a fix-up to the commit that added the `read_message` functionality to the Queue class. see c7e02846bc6954be4cc9e9d6999bba3acdd149ca